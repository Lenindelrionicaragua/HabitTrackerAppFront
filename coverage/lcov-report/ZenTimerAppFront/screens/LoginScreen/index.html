
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for ZenTimerAppFront/screens/LoginScreen</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> ZenTimerAppFront/screens/LoginScreen</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">74.48% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>73/98</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">69.44% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>25/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.19% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>16/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">74.48% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>73/98</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="LoginScreen.js"><a href="LoginScreen.js.html">LoginScreen.js</a></td>
	<td data-value="65.75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 65%"></div><div class="cover-empty" style="width: 35%"></div></div>
	</td>
	<td data-value="65.75" class="pct medium">65.75%</td>
	<td data-value="73" class="abs medium">48/73</td>
	<td data-value="60.71" class="pct medium">60.71%</td>
	<td data-value="28" class="abs medium">17/28</td>
	<td data-value="70.58" class="pct medium">70.58%</td>
	<td data-value="17" class="abs medium">12/17</td>
	<td data-value="65.75" class="pct medium">65.75%</td>
	<td data-value="73" class="abs medium">48/73</td>
	</tr>

<tr>
	<td class="file high" data-value="LoginScreenStyles.js"><a href="LoginScreenStyles.js.html">LoginScreenStyles.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-06-08T19:38:16.357Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    