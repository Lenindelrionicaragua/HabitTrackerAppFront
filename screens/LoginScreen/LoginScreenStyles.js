import styled from "styled-components/native";
import { View, Image, Text, TextInput, Pressable } from "react-native";
import { Colors } from "../../styles/AppStyles";
import { SafeAreaView } from "react-native";

const { white, orange, grey, yellow, lightGrey, black } = Colors;

const green = "#00ff00";
const red = "#ff0000";

export const StyledContainer = styled(SafeAreaView)`
  flex: 1;
  background-color: ${lightGrey};
  width: 100%;
`;

export const InnerContainer = styled(View)`
  background-color: ${lightGrey};
  flex: 1;
  padding-top: 15px;
  width: 100%;
  align-items: center;
  justify-content: center;
`;

export const PageLogo = styled(Image)`
  width: 160px;
  height: 160px;
  border-radius: 80px;
  overflow: hidden;
`;

export const PageTitle = styled(Text)`
  font-size: 30px;
  text-align: center;
  font-weight: bold;
  color: ${black};
  padding: 10px;
`;

export const SubTitle = styled(Text)`
  font-size: 10px;
  margin-bottom: 20px;
  letter-spacing: 1px;
  font-weight: bold;
  color: ${grey};
`;

export const StyledFormArea = styled(View)`
  width: 90%;
`;

export const StyledTextInput = styled(TextInput)`
  background-color: ${orange};
  padding: 15px;
  padding-left: 55px;
  padding-right: 55px;
  border-radius: 5px;
  font-size: 16px;
  height: 60px;
  margin-vertical: 3px;
  margin-bottom: 10px;
  color: ${grey};
`;

export const StyledInputLabel = styled(Text)`
  color: ${grey};
  font-size: 13px;
  text-align: left;
`;

export const LeftIcon = styled(View)`
  left: 15px;
  top: 38px;
  position: absolute;
  z-index: 1;
`;

export const RightIcon = styled(Pressable)`
  right: 15px;
  top: 38px;
  position: absolute;
  z-index: 1;
`;

export const StyledButton = styled(Pressable)`
  padding: 15px;
  background-color: ${black};
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  margin-vertical: 5px;
  height: 60px;

  ${props =>
    props.google == true &&
    `
    background-color: ${orange};
    flex-direction: row;
    justify-content: center;
  `}
`;

export const ButtonText = styled(Text)`
  color: ${props => (props.google ? black : white)};
  font-size: 16px;

  ${props =>
    props.google == true &&
    `
    padding-left: 15px;
  `}
`;

export const MsgBox = styled(Text)`
  text-align: center;
  font-size: 13px;
  color: ${prop => (prop.type === "SUCCESS" ? green : red)};
`;

export const Line = styled(View)`
  height: 1px;
  width: 100%;
  background-color: ${orange};
  margin-vertical: 10px;
`;

export const FooterView = styled(View)`
  justify-content: center;
  flex-direction: row;
  align-items: center;
  padding: 10px;
`;

export const FooterText = styled(Text)`
  justify-content: center;
  align-content: center;
  color: ${grey};
  font-size: 15px;
`;

export const SignupLink = styled(Pressable)`
  justify-content: center;
  align-items: center;
`;

export const SignupLinkContent = styled(Text)`
  color: ${yellow};
  font-size: 15px;
`;
