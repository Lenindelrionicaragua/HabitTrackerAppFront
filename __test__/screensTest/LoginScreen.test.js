import React from "react";
import { render, fireEvent, act, cleanup } from "@testing-library/react-native";
import MockAdapter from "axios-mock-adapter";
import axios from "axios";
import renderer from "react-test-renderer";
import { Fontisto } from "@expo/vector-icons";
import LoginScreen from "../../screens/LoginScreen/LoginScreen";
import { Formik } from "formik";
import { StatusBar } from "react-native";
import { PageLogo } from "../../screens/LoginScreen/LoginScreenStyles";

// Rendering Functions
const renderLoginScreen = () => render(<LoginScreen />);
const renderLoginScreenWithRenderer = () => renderer.create(<LoginScreen />);

let loginScreenRender;
let loginScreenRenderWithRenderer;

beforeEach(async () => {
  loginScreenRender = renderLoginScreen();
  loginScreenRenderWithRenderer = renderLoginScreenWithRenderer();
});

//LoginScreen
describe("LoginScreen", () => {
  afterEach(() => {
    cleanup();
  });

  test("Renders correctly the LoginScreen Component", () => {
    const loginScreenSnapshot = loginScreenRenderWithRenderer.toJSON();
    expect(loginScreenSnapshot).toMatchSnapshot();
  });

  test("Render StatusBar correctly", () => {
    const loginScreenInstance = loginScreenRenderWithRenderer.root;
    const statusBar = loginScreenInstance.findByType(StatusBar);
    expect(statusBar).toBeTruthy();
  });

  test("Render UI components correctly", () => {
    const { getByTestId } = loginScreenRender;

    expect(getByTestId("styled-container")).toBeTruthy();
    expect(getByTestId("inner-container")).toBeTruthy();
    expect(getByTestId("page-logo")).toBeTruthy();
    expect(getByTestId("page-title")).toBeTruthy();
    expect(getByTestId("sub-title")).toBeTruthy();
  });

  test("The PageLogo must have a valid image source", () => {
    const { getByTestId } = loginScreenRender;
    const pageLogoComponent = getByTestId("page-logo");
    expect(pageLogoComponent.props.source).toBeTruthy();
  });

  test("PageTitle should render a string of letters, numbers, or spaces", () => {
    const { getByTestId } = loginScreenRender;
    const pageTitleComponent = getByTestId("page-title");
    const textContent = pageTitleComponent.props.children.toString();
    expect(textContent).toMatch("ZenTimer");
  });

  test("SubTitle should render a string of letters, numbers or spaces", () => {
    const { getByTestId } = loginScreenRender;
    const subTitleComponent = getByTestId("sub-title");
    const textContent = subTitleComponent.props.children.toString();
    expect(textContent).toMatch("Account Login");
  });
});

// PageLogo
describe("PageLogo", () => {
  afterEach(() => {
    cleanup();
  });

  it("Renders the PageLogo component correctly", () => {
    const pageLogoSnapshot = renderer.create(<PageLogo />).toJSON();
    expect(pageLogoSnapshot).toMatchSnapshot();
  });
});

// Formik Integration Tests
describe("Formik Integration Tests", () => {
  let formikComponent;

  beforeEach(() => {
    const loginScreenInstance = loginScreenRenderWithRenderer.root;
    formikComponent = loginScreenInstance.findByType(Formik);
  });

  afterEach(() => {
    cleanup();
  });

  test("Render a Formik component", () => {
    expect(formikComponent).toBeTruthy();
  });

  test("Should have initialValues", () => {
    expect(formikComponent.props.initialValues).toEqual({
      email: "",
      password: ""
    });
  });

  test("Formik should have onSubmit function", () => {
    expect(formikComponent.props.onSubmit).toBeDefined();
  });

  test("Formik should handle onSubmit", () => {
    expect(formikComponent.props.onSubmit).toBeInstanceOf(Function);
  });
});

// LoginTextInput
describe("LoginTextInput", () => {
  let emailInput;
  let passwordInput;

  const renderForm = () => {
    const { getByTestId } = loginScreenRender;
    emailInput = getByTestId("email-input");
    passwordInput = getByTestId("password-input");
  };

  beforeEach(() => {
    renderForm();
  });

  afterEach(() => {
    cleanup();
  });

  describe("Rendering", () => {
    test("Renders correctly the email-input", () => {
      expect(emailInput).toBeTruthy();
    });

    test("Renders correctly the password-input", () => {
      expect(passwordInput).toBeTruthy();
    });
  });

  describe("Form State Update", () => {
    beforeEach(() => {
      renderForm();
    });

    afterEach(() => {
      cleanup();
    });

    test("Correctly updates form state on onChangeText and onBlur", () => {
      const { getByTestId } = loginScreenRender;

      act(() => {
        fireEvent.changeText(getByTestId("email-input"), "serenity@gmail.com");
        fireEvent(getByTestId("email-input"), "blur", {
          target: { value: "serenity@gmail.com" }
        });

        fireEvent.changeText(getByTestId("password-input"), "password123");
        fireEvent(getByTestId("password-input"), "blur", {
          target: { value: "password123" }
        });
      });

      expect(getByTestId("email-input").props.value).toBe("serenity@gmail.com");
      expect(getByTestId("password-input").props.value).toBe("password123");
    });
  });

  // Login StyledButton
  describe("StyledButton", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render StyledButton", () => {
      const { getByTestId } = loginScreenRender;
      const styledButtonElement = getByTestId("login-styled-button");
      expect(styledButtonElement).toBeTruthy();
    });
  });

  // MsgBox
  describe("MsgBox", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render a MsgBox", () => {
      const { getByTestId } = loginScreenRender;
      const msgBoxElement = getByTestId("msg-box");
      expect(msgBoxElement).toBeTruthy();
    });

    test("MsgBox should render a string of letters, numbers or spaces", async () => {
      const { getByTestId } = loginScreenRender;
      const msgBoxElement = getByTestId("msg-box");
      const textContent = msgBoxElement.props.children.toString();
      expect(textContent).toMatch(/^[a-zA-Z0-9.\s]*$/);
    });
  });

  describe("Google StyledButton", () => {
    afterEach(() => {
      cleanup();
    });

    // Google StyledButton
    test("Render correctly", () => {
      const { getByTestId } = loginScreenRender;
      const googleStyledButton = getByTestId("google-styled-button");
      expect(googleStyledButton).toBeTruthy();
    });

    test("Render the Google Icon", () => {
      const { getByTestId } = loginScreenRender;
      const googleIconElement = getByTestId("google-icon");
      expect(googleIconElement).toBeTruthy();
    });

    test("Render the correct text", () => {
      const { getByTestId } = loginScreenRender;
      const googleButtonTextElement = getByTestId("google-button-text");
      const textContent = googleButtonTextElement.props.children.toString();
      expect(textContent).toMatch("Sign in with Google");
    });

    test("StyledButton should have an Fontisto component as Child", () => {
      const { getByTestId } = loginScreenRender;
      const styledButtonComponent = getByTestId("google-styled-button");
      const children = styledButtonComponent.props.children;

      let hasFontistoAsAChild = false;

      React.Children.forEach(children, child => {
        if (child && child.type === Fontisto) {
          hasFontistoAsAChild = true;
        }
      });
      expect(hasFontistoAsAChild).toBe(true);
    });

    test("Render a google-button-text", () => {
      const { getByTestId } = loginScreenRender;
      const buttonTextGoogle = getByTestId("google-button-text");
      expect(buttonTextGoogle).toBeTruthy();
    });
  });

  // Footer
  describe("FooterView", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render correctly", () => {
      const { getByTestId } = loginScreenRender;
      const footerViewElement = getByTestId("footer-view");
      expect(footerViewElement).toBeTruthy();
    });
  });

  describe("FooterText", () => {
    afterEach(() => {
      cleanup();
    });

    test("Render correctly", () => {
      const { getByTestId } = loginScreenRender;
      const footerTextElement = getByTestId("footer-text");
      expect(footerTextElement).toBeTruthy();
    });

    test("Render a text", () => {
      const { getByTestId } = loginScreenRender;
      const footerTextElement = getByTestId("footer-text");
      const textContent = footerTextElement.props.children;
      expect(textContent).toBe("Dont you have an account already?");
    });
  });

  describe("SignupLink", () => {
    test("Render correctly", () => {
      const { getByTestId } = loginScreenRender;
      const SignupLinkElement = getByTestId("signup-link");
      expect(typeof SignupLinkElement.props).toBe("object");
    });

    test("Render the correct text", () => {
      const { getByTestId } = loginScreenRender;
      const SignupLinkContent = getByTestId("signup-link-content");
      const LinkContent = SignupLinkContent.props.children;
      expect(LinkContent).toBe("Signup");
    });
  });

  //Navigation Test

  describe("loginScreen navigation", () => {
    let loginScreenInstance;
    let navigation;

    beforeEach(() => {
      navigation = { navigate: jest.fn() };

      const loginScreenRenderWithRenderer = renderer.create(
        <LoginScreen navigation={navigation} />
      );
      loginScreenInstance = loginScreenRenderWithRenderer.root;
    });

    afterEach(() => {
      cleanup();
    });

    test("Navigate to SignupScreen when Signup link is clicked", () => {
      const { getByTestId } = render(<LoginScreen navigation={navigation} />);
      const signupLink = getByTestId("signup-link");

      act(() => {
        fireEvent.press(signupLink);
      });

      expect(navigation.navigate).toHaveBeenCalledWith("SignupScreen");
    });

    test("navigate to WelcomeScreen when Login button is clicked", async () => {
      const mock = new MockAdapter(axios);
      const formikComponent = loginScreenInstance.findByType(Formik);
      const handleLogin = formikComponent.props.onSubmit;
      const setSubmitting = jest.fn();

      const url =
        "https://zen-timer-app-server-7f9db58def4c.herokuapp.com/api/auth/log-in";

      mock.onPost(url).reply(200, {
        success: true,
        msg: "Login successful",
        user: {
          id: "6638d16d96bf8c3d0d4cf2e4",
          email: "johnlenin@email.com",
          name: "John Lenin"
        }
      });

      await act(async () => {
        await handleLogin(
          { email: "johnlenin@email.com", password: "Password1234!" },
          { setSubmitting }
        );
      });

      expect(navigation.navigate).toHaveBeenCalledWith("WelcomeScreen", {
        id: "6638d16d96bf8c3d0d4cf2e4",
        email: "johnlenin@email.com",
        name: "John Lenin"
      });

      expect(setSubmitting).toHaveBeenCalledWith(false);
    });

    test("navigate to WelcomeScreen should fail if request does not have a valid email", async () => {
      const mock = new MockAdapter(axios);
      const formikComponent = loginScreenInstance.findByType(Formik);
      const handleLogin = formikComponent.props.onSubmit;
      const setSubmitting = jest.fn();

      const url =
        "https://zen-timer-app-server-7f9db58def4c.herokuapp.com/api/auth/log-in";

      mock.onPost(url).reply(200, {
        success: false,
        error: "BAD REQUEST: Email and password are required."
      });

      await act(async () => {
        await handleLogin(
          { email: "", password: "Password1234!" },
          { setSubmitting }
        );
      });

      expect(navigation.navigate).not.toHaveBeenCalled();
      expect(setSubmitting).toHaveBeenCalledWith(false);
    });

    test("navigate to WelcomeScreen should fail if request does not have a valid password", async () => {
      const mock = new MockAdapter(axios);
      const formikComponent = loginScreenInstance.findByType(Formik);
      const handleLogin = formikComponent.props.onSubmit;
      const setSubmitting = jest.fn();

      const url =
        "https://zen-timer-app-server-7f9db58def4c.herokuapp.com/api/auth/log-in";

      mock.onPost(url).reply(200, {
        success: false,
        error: "BAD REQUEST: Email and password are required."
      });

      await act(async () => {
        await handleLogin(
          { email: "johnlenin@email.com", password: "" },
          { setSubmitting }
        );
      });

      expect(navigation.navigate).not.toHaveBeenCalled();
      expect(setSubmitting).toHaveBeenCalledWith(false);
    });

    test("navigate to WelcomeScreen should fail if request is an empty object", async () => {
      const mock = new MockAdapter(axios);
      const formikComponent = loginScreenInstance.findByType(Formik);
      const handleLogin = formikComponent.props.onSubmit;
      const setSubmitting = jest.fn();

      const url =
        "https://zen-timer-app-server-7f9db58def4c.herokuapp.com/api/auth/log-in";

      mock.onPost(url).reply(200, {
        success: false,
        msg: "Invalid request body"
      });

      await act(async () => {
        await handleLogin({}, { setSubmitting });
      });

      expect(navigation.navigate).not.toHaveBeenCalled();
      expect(setSubmitting).toHaveBeenCalledWith(false);
    });
  });
});
