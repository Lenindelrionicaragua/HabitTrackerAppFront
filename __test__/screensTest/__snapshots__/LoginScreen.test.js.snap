// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Renders correctly the LoginScreen Component 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
  testID="keyboard-avoiding-view"
>
  <RCTScrollView
    testID="scroll-view"
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="pressable"
      >
        <RCTSafeAreaView
          style={
            {
              "backgroundColor": "#f9f7f3",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "width": "100%",
            }
          }
          testID="styled-container"
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#f9f7f3",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "paddingTop": 15,
                "width": "100%",
              }
            }
            testID="inner-container"
          >
            <Image
              resizeMode="cover"
              source={1}
              style={
                {
                  "borderBottomLeftRadius": 80,
                  "borderBottomRightRadius": 80,
                  "borderTopLeftRadius": 80,
                  "borderTopRightRadius": 80,
                  "height": 160,
                  "overflow": "hidden",
                  "width": 160,
                }
              }
              testID="page-logo"
            />
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                  "textAlign": "center",
                }
              }
              testID="page-title"
            >
              ZenTimer
            </Text>
            <Text
              style={
                {
                  "color": "#1a1204",
                  "fontSize": 10,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "marginBottom": 20,
                }
              }
              testID="sub-title"
            >
              Account Login
            </Text>
            <View
              style={
                {
                  "width": "90%",
                }
              }
            >
              <View
                testID="text-input-login-screen"
              >
                <View
                  style={
                    {
                      "left": 15,
                      "position": "absolute",
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="left-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                    testID="octicons-icon"
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#1a1204",
                      "fontSize": 13,
                      "textAlign": "left",
                    }
                  }
                >
                  Email Address
                </Text>
                <TextInput
                  keyboardType="email-address"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="serenity@gmail.com"
                  placeholderTextColor="#f9f7f3"
                  style={
                    {
                      "backgroundColor": "#ffc500",
                      "borderBottomLeftRadius": 5,
                      "borderBottomRightRadius": 5,
                      "borderTopLeftRadius": 5,
                      "borderTopRightRadius": 5,
                      "color": "#1a1204",
                      "fontSize": 16,
                      "height": 60,
                      "marginBottom": 10,
                      "marginVertical": 3,
                      "paddingBottom": 15,
                      "paddingLeft": 55,
                      "paddingRight": 55,
                      "paddingTop": 15,
                    }
                  }
                  testID="email-input"
                  value=""
                />
              </View>
              <View
                testID="text-input-login-screen"
              >
                <View
                  style={
                    {
                      "left": 15,
                      "position": "absolute",
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="left-icon"
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 30,
                        },
                        undefined,
                        {
                          "fontFamily": "octicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                    testID="octicons-icon"
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    {
                      "color": "#1a1204",
                      "fontSize": 13,
                      "textAlign": "left",
                    }
                  }
                >
                  Password
                </Text>
                <TextInput
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="* * * * * * *"
                  placeholderTextColor="#f9f7f3"
                  secureTextEntry={true}
                  style={
                    {
                      "backgroundColor": "#ffc500",
                      "borderBottomLeftRadius": 5,
                      "borderBottomRightRadius": 5,
                      "borderTopLeftRadius": 5,
                      "borderTopRightRadius": 5,
                      "color": "#1a1204",
                      "fontSize": 16,
                      "height": 60,
                      "marginBottom": 10,
                      "marginVertical": 3,
                      "paddingBottom": 15,
                      "paddingLeft": 55,
                      "paddingRight": 55,
                      "paddingTop": 15,
                    }
                  }
                  testID="password-input"
                  value=""
                />
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "position": "absolute",
                      "right": 15,
                      "top": 38,
                      "zIndex": 1,
                    }
                  }
                  testID="right-icon"
                >
                  <Text
                    allowFontScaling={false}
                    onPress={[Function]}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#f9f7f3",
                          "fontSize": 25,
                        },
                        undefined,
                        {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                    testID="feather-icon"
                  >
                    
                  </Text>
                </View>
              </View>
              <Text
                style={
                  {
                    "color": "#ff0000",
                    "fontSize": 13,
                    "textAlign": "center",
                  }
                }
                testID="msg-box"
                type="ERROR"
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#000000",
                    "borderBottomLeftRadius": 5,
                    "borderBottomRightRadius": 5,
                    "borderTopLeftRadius": 5,
                    "borderTopRightRadius": 5,
                    "height": 60,
                    "justifyContent": "center",
                    "marginVertical": 5,
                    "paddingBottom": 15,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 15,
                  }
                }
                testID="login-styled-button"
              >
                <Text
                  style={
                    {
                      "color": "#ffe1a8",
                      "fontSize": 16,
                    }
                  }
                  testID="login-button-text"
                >
                  Login
                </Text>
              </View>
              <View
                style={
                  {
                    "backgroundColor": "#ffc500",
                    "height": 1,
                    "marginVertical": 10,
                    "width": "100%",
                  }
                }
                testID="line"
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                google={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#ffc500",
                    "borderBottomLeftRadius": 5,
                    "borderBottomRightRadius": 5,
                    "borderTopLeftRadius": 5,
                    "borderTopRightRadius": 5,
                    "flexDirection": "row",
                    "height": 60,
                    "justifyContent": "center",
                    "marginVertical": 5,
                    "paddingBottom": 15,
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 15,
                  }
                }
                testID="google-styled-button"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#1a1204",
                        "fontSize": 20,
                      },
                      undefined,
                      {
                        "fontFamily": "Fontisto",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                  testID="google-icon"
                >
                  
                </Text>
                <Text
                  google={true}
                  style={
                    {
                      "color": "#000000",
                      "fontSize": 16,
                      "paddingLeft": 15,
                    }
                  }
                  testID="google-button-text"
                >
                  Sign in with Google
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
                testID="footer-view"
              >
                <Text
                  style={
                    {
                      "alignContent": "center",
                      "color": "#1a1204",
                      "fontSize": 15,
                      "justifyContent": "center",
                    }
                  }
                  testID="footer-text"
                >
                  Dont you have an account already?
                </Text>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                  testID="signup-link"
                >
                  <Text
                    style={
                      {
                        "color": "#ffc554",
                        "fontSize": 15,
                      }
                    }
                    testID="signup-link-content"
                  >
                    Signup
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`PageLogo Renders the PageLogo component correctly 1`] = `
<Image
  style={
    {
      "borderBottomLeftRadius": 80,
      "borderBottomRightRadius": 80,
      "borderTopLeftRadius": 80,
      "borderTopRightRadius": 80,
      "height": 160,
      "overflow": "hidden",
      "width": 160,
    }
  }
/>
`;
